sympy/matrices/common.py

```diff
@@ -2420,16 +2444,22 @@ def pow(self, exp, jordan=None, dotprodsimp=None):
             # When certain conditions are met,
             # Jordan block algorithm is faster than
             # computation by recursion.
-            elif jordan_pow is not None and (jordan or \
-                    (jordan is not False and a.rows == 2 and exp > 100000)):
+            elif jordan_pow is not None and (method == 'jordan' or \
+                    (a.rows == 2 and exp > 100000)):
                 try:
                     return jordan_pow(exp)
                 except MatrixError:
-                    if jordan:
+                    if method == 'jordan':
                         raise
 
+
```

Comment:
The extra blank line here is not needed.

I think that the logic of this function should be rearranged so it looks like:
```
if method == 'multiply':
    #
elif method == 'mulsimp':
    ...
elif method is None:
    #
else:
    raise
```